addons/website_sale_stock/models/product_product.py

```diff
@@ -1,19 +1,49 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import models, fields, _
 from odoo.http import request
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    stock_notification_partner_ids = fields.Many2many('res.partner', string='Back in stock Notifications')
+
+    def has_stock_notification(self, partner_id):
+        self.ensure_one()
+
+        return partner_id in self.stock_notification_partner_ids
+
     def _get_cart_qty(self, website=None):
         if not self.allow_out_of_stock_order:
             website = website or self.env['website'].get_current_website()
-            cart = website and request and website.sale_get_order() or None
+            cart = website and request and hasattr(request, 'website') and website.sale_get_order() or None
```

Comment:
I remember this is a bug fix, but I don't remember what causes the crash, don't you think we need to backport this?

Reply:
Basically when the cron is run manually, `request` has no attribute `website`, and that would cause a crash

Reply:
Alright thanks, would be nice to have it as a comment above, since I had forgotten it 😄

Reply:
All right!